client/lib/products-values/index.js

```diff
@@ -42,7 +42,8 @@ function formatProduct( product ) {
 		is_domain_registration: product.is_domain_registration !== undefined
 			? product.is_domain_registration
 			: product.isDomainRegistration,
-		free_trial: product.free_trial || product.freeTrial
+		free_trial: product.free_trial || product.freeTrial,
+		is_email_verified: Boolean( product.is_email_verified )
```

Comment:
Is this actually needed? Looking at the backend changes, `is_email_verified` can only be a boolean. And you do not need to change its name (like the other properties here). So I think it's safe to remove this line altogether. `assign` will just copy the field over from `product`.